src/gui/sdlmain.cpp

```diff
@@ -3499,28 +3554,20 @@ static void HandleMouseButton(SDL_MouseButtonEvent * button) {
 			break; // Don't pass click to mouse handler
 		}
 		switch (button->button) {
-		case SDL_BUTTON_LEFT:
-			Mouse_ButtonPressed(0);
-			break;
-		case SDL_BUTTON_RIGHT:
-			Mouse_ButtonPressed(1);
-			break;
-		case SDL_BUTTON_MIDDLE:
-			Mouse_ButtonPressed(2);
-			break;
+		case SDL_BUTTON_LEFT:   MOUSE_EventPressed(0); break;
+		case SDL_BUTTON_RIGHT:  MOUSE_EventPressed(1); break;
+		case SDL_BUTTON_MIDDLE: MOUSE_EventPressed(2); break;
+		case SDL_BUTTON_X1:     MOUSE_EventPressed(3); break;
+		case SDL_BUTTON_X2:     MOUSE_EventPressed(4); break;
 		}
 		break;
 	case SDL_RELEASED:
 		switch (button->button) {
-		case SDL_BUTTON_LEFT:
-			Mouse_ButtonReleased(0);
-			break;
-		case SDL_BUTTON_RIGHT:
-			Mouse_ButtonReleased(1);
-			break;
-		case SDL_BUTTON_MIDDLE:
-			Mouse_ButtonReleased(2);
-			break;
+		case SDL_BUTTON_LEFT:   MOUSE_EventReleased(0); break;
+		case SDL_BUTTON_RIGHT:  MOUSE_EventReleased(1); break;
+		case SDL_BUTTON_MIDDLE: MOUSE_EventReleased(2); break;
+		case SDL_BUTTON_X1:     MOUSE_EventReleased(3); break;
+		case SDL_BUTTON_X2:     MOUSE_EventReleased(4); break;
```

Comment:
SDL defines:
```c++
SDL_mouse.h-#define SDL_BUTTON_LEFT     1
SDL_mouse.h-#define SDL_BUTTON_MIDDLE   2
SDL_mouse.h:#define SDL_BUTTON_RIGHT    3
SDL_mouse.h-#define SDL_BUTTON_X1       4
SDL_mouse.h-#define SDL_BUTTON_X2       5
```

Both of the nested switches could be eliminated:

```c++
switch (button->state)
case SDL_PRESSED:
	...
	MOUSE_EventPressed(button->button);
	break;
case SDL_RELEASED:
	MOUSE_EventReleased(button->button);
	break;
```

Then in:

``` c++
void MOUSE_EventReleased(const int event_id)
{
        switch (event_id) {
        case 1: // left button
        case 2: // middle button
        case 3: // right button
        case 4: // extra button 1
        case 5: // extra button 2
```

Or a bit more SDL-specific:

``` c++
void MOUSE_EventReleased(const int event_id)
{
        switch (event_id) {
        case SDL_BUTTON_LEFT:
        case SDL_BUTTON_MIDDLE:
        case SDL_BUTTON_RIGHT:
        case SDL_BUTTON_X1:
        case SDL_BUTTON_X2:
```

(or some hybrid where we map thee to our own mouse-enum, if this function will eventually be accessed by the VMWare driver? 

Just some thoughts, if you think it helps. Feel free to disregard if it's a bad fit down the road.

Reply:
Unfortunately, this SDL numbering scheme is not practical for us, for several reasons:
- the button index is also used as array index (for `mouse.last_pressed_x[]` and relevant), so it should start from 0, not from 1
- DOS mouse driver (INT 33) contains two functions (0x05 and 0x06) that also use 0 for left, 1 for right, and 2 for middle button as indices in `reg_bx` - so it is convenient to use the same scheme, using SDL-compatible button id's would have complicated implementation there
- left and right button are supported by every emulated mouse, unlike middle, and X1/X2 buttons are supported by IntelliMouse Explorer mouse only (not yet emulated, but planned); with current index scheme we can always easily check whether button is interesting for the particular mouse with simple comparing against constant

So I prefer to keep it the current way.